### PR TITLE
Fix GCC 13 and Python3.11 Compilation Errors

### PR DESCRIPTION
**Pull request by yameatmeyourdead**
Implements the fixes described in https://github.com/m2-farzan/ros2-galactic-PKGBUILD/issues/38

### CODE REVIEW

**Comment by yameatmeyourdead** [comment]
Can do one better, my PR was merged in rosbag2 so patch file is no longer needed

File: PKGBUILD [hunk 1]
```diff
@@ -77,6 +77,22 @@ prepare() {
...
+    ## rclcpp: Fix missing stdexcept includes
+    git -C $srcdir/ros2/src/ros2/rclcpp cherry-pick 86c77143c96d85711a87f2a5adcc4d7f0fb0dbeb
+    ## pybind11_vendor: Support for python 3.11
+    git -C $srcdir/ros2/src/ros2/pybind11_vendor checkout 3.0.3
+    ## rosbag2_storage: cherry pick to fix missing cstdint include
+    git -C $srcdir/ros2/src/ros2/rosbag2 cherry-pick 5f5a583ca2e47f6a2ec1bca6076576a99efaaf73
+    ## rosbag2_compression: cherry pick to fix missing cstdint include
+    git -C $srcdir/ros2/src/ros2/rosbag2 cherry-pick 65c889e1fa55dd85a148b27b8c27dadc73238e67
+    ## image_pipeline: revert broken commit
+    git -C $srcdir/ros2/src/ros-visualization/image_pipeline revert 46f4a1849843efee368415b95498e7b7b7ece03b
```
Comment (lprobsth) [inline PKGBUILD:95]: Can you clarify why image_pipeline is included?
This part fails for me and I don't see when image_pipeline was ever included in ros2 galactic.

**Comment by yameatmeyourdead** [comment]
Sorry, never finished this. I will build this and print SRCINFO to get this PR merged.

File: PKGBUILD (hunk 1 above)
Comment (yameatmeyourdead) [reply PKGBUILD:95]: Good question, I honestly have no idea. I remember this working a long time ago when I started (and apparently never finished) this PR but it does fail for me as well now.

**Comment by yameatmeyourdead** [comment]
Successfully built on 6.1.55-1-MANJARO with GCC 13.2.1 and python 3.11.5
SRCINFO has been updated.
Original .patch file that was talked about in first post is no longer required, upstream has commit that fixes.
Please let me know if you need anything else.

**Review by m2-farzan — APPROVED** [review]

**Comment by m2-farzan** [comment]
Thank you for your contribution!

**Pull request merged**